File: src/main/java/com/destino/projeto_destino/dto/compra/ViagemDetalhadaDTO.java
package com.destino.projeto_destino.dto.compra;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

public record ViagemDetalhadaDTO(
        Long id,
        String nomePacote,
        String descricao,
        BigDecimal valor,
        String statusCompra,
        @JsonFormat(pattern = "dd-MM-yyyy")
        LocalDate dataPartida,
        @JsonFormat(pattern = "dd-MM-yyyy")
        LocalDate dataRetorno,
        @JsonFormat(pattern = "dd-MM-yyyy")
        LocalDate dataCompra,
        String numeroReserva,
        String imagemPrincipal,
        List<String> galeria,
        List<String> inclusoes,
        String nomeHotel,
        String tipoTransporte
) {}

File: src/main/kotlin/com/fatec/destino/dto/viagem/ViagemDetalhadaDTO.kt
package com.fatec.destino.dto.viagem

import com.fatec.destino.model.pacote.pacoteFoto.foto.Foto
import com.fatec.destino.util.model.transporte.Meio
import java.math.BigDecimal
import java.time.LocalDate
import java.util.*

data class ViagemDetalhadaDTO(
    val id: UUID?,
    val nomePacote: String?,
    val descricao: String?,
    val valor: BigDecimal?,
    val statusCompra: String?,
    val dataPartida: LocalDate?,
    val dataRetorno: LocalDate?,
    val dataCompra: Date,
    val numeroReserva: String?,
    val imagemPrincipal: String?,
    val galeria: List<Foto?>?,
    val inclusoes: List<String>?,
    val nomeHotel: String?,
    val tipoTransporte: Meio?
)
